tests/controllers/villains.test.js

```diff
@@ -0,0 +1,141 @@
+      await getAllVillains({}, response)
+      expect(stubbedFindAll).to.have.callCount(1)
+      expect(stubbedStatus).to.have.been.calledWith(500)
+      expect(stubbedStatusDotSend).to.have.been.calledWith('Unable to retrieve all villains try again.')
+    })
+  })
+  describe('getVillainBySlug', () =>
+  {
+    it('Retrieves villain associated with the provided slug from the database and calls response.send ', async () =>
+    {
+      stubbedFindOne.returns(singleVillain)
+      const request = { params: { slug: 'hades' } }
+
+      await getVillainBySlug(request, response)
+      expect(stubbedFindOne).to.be.calledWith({
+        where: { slug: 'hades' },
+        attributes: ['name', 'movie', 'slug']
+      })
+      expect(stubbedStatus).to.have.been.calledWith(200)
+      expect(stubbedStatusDotSend).to.have.been.calledWith(singleVillain)
+    })
+    it('Returns a 404 status and a message when no villain is found', async () =>
+    {
+      stubbedFindOne.returns(null)
+      const request = { params: { slug: 'ursula' } }
+
+      await getVillainBySlug(request, response)
+      expect(stubbedFindOne).to.be.calledWith({
+        where: { slug: 'ursula' },
+        attributes: ['name', 'movie', 'slug']
+      })
+      expect(stubbedStatus).to.have.been.calledWith(404)
+      expect(stubbedStatusDotSend).to.have.been.calledWith('It looks like ursula could not be found')
+    })
+    it('Returns a 500 status with a message when the database call throws an error.', async () =>
+    {
+      stubbedFindOne.throws('ERROR!')
+      await getVillainBySlug({}, response)
+      expect(stubbedFindOne).to.have.callCount(0)
+      expect(stubbedStatus).to.have.been.calledWith(500)
+      expect(stubbedStatusDotSend).to.have.been.calledWith('Unable to retrieve the villain, please try again.')
+    })
+  })
+  describe('saveNewVillain', () =>
+  {
+    it('Saves a new villain to the database and responds with a 201 status and sends a message with it.', async () =>
+    {
+      stubbedCreate.returns(singleVillain)
+      const request = { body: singleVillain }
+
+      await saveNewVillain(request, response)
+      expect(stubbedCreate).to.be.calledWith(singleVillain)
+      expect(stubbedStatus).to.have.been.calledWith(201)
+      expect(stubbedStatusDotSend).to.have.been.calledWith(singleVillain)
+    })
+    it('Returns a 400 status and a message when no villain is found.', async () =>
+    {
+      stubbedCreate.returns(null)
+      const request = { body: {} }
+
+      await saveNewVillain(request, response)
+      expect(stubbedCreate).to.have.callCount(0)
+      expect(stubbedStatus).to.have.been.calledWith(400)
+      expect(stubbedStatusDotSend).to.have.been.calledWith('The required fields are name, movie, and slug.')
+    })
+    it('Returns a 500 status and a message when the database call throws an error.', async () =>
+    {
+      stubbedCreate.throws('ERROR!')
+      await saveNewVillain({}, response)
```

Comment:
This test is failing because you are sending in an empty request object but the code expects there to be a body with all the fields in it. Without that you would never get to the line that calls create for that function to throw an error which triggers the catch code. Of course your code doesn't have error handling for this function so it wouldn't work anyway.